core/debug/src/debug.c

```diff
@@ -230,3 +230,123 @@ void debug_fault(THaltError reason, uint32_t lr, uint32_t sp)
     debug_mpu_config();
     DEBUG_PRINT_END();
 }
+
+static void __debug_reboot(void)
+{
+    UVISOR_SVC(UVISOR_SVC_ID_DEBUG_REBOOT, "");
+}
+
+/* This function must be called as an SVCall handler.
+ * All debug handlers that are required to reboot upon exit should use the
+ * __debug_reboot function as return value, which triggers the SVCall. This note
+ * applies to uVisor internally, as the actual debug box does not need to care
+ * about this. */
+void debug_reboot(void)
+{
+    if (!g_debug_box.initialized || g_active_box != g_debug_box.box_id) {
+        HALT_ERROR(NOT_ALLOWED, "This function can only be called from the context of an initialized debug box.\n\r");
```

Comment:
Please add a FIXME for returning an error code in case an unauthorised wrong box calls this function. Potentially add a JIRA backlog-ticket (Q3?) for a reset-ACL to allow individual boxes to reset the system.